docs: improve consuming partial-Ivy code outside the Angular CLI

alan-agius4 opened the pull request:
With this change we add an example of how to consume the linker plugin and also clean up the parts of this section.

Closes #44026

alan-agius4 committed:
docs: improve consuming partial-Ivy code outside the Angular CLI

With this change we add an example of how to consume the linker plugin and also clean up the parts of this section.

Closes #44026

mary-poppins commented:
You can preview a500fd8 at https://pr44027-a500fd8.ngbuilds.io/.

petebacondarwin reviewed (changes requested):
Looks good @alan-agius4.
We are trying to avoid adding inline code snippets to the docs.
Do you need any help moving this to an external example?

mary-poppins commented:
You can preview 8028832 at https://pr44027-8028832.ngbuilds.io/.

mary-poppins commented:
You can preview 92471a9 at https://pr44027-92471a9.ngbuilds.io/.

mary-poppins commented:
You can preview 3472e54 at https://pr44027-3472e54.ngbuilds.io/.

petebacondarwin reviewed (approved):
Great work @alan-agius4 thanks.
Strictly the second commit should probably be a `build(docs-infra): ...` but won't block the merge over it.

gkalpak commented on aio/tools/transforms/examples-package/services/region-parser.js:
```diff
@@ -10,6 +10,7 @@ module.exports = function regionParser() {
   regionParserImpl.regionMatchers = {
     ts: inlineC,
     js: inlineC,
+    mjs: inlineCOnly,
```
OOC, why not `inlineC`?

alan-agius4 replied:
Because of the docs plaster comment, looks neater using `// …` IMHO.

alan-agius4 committed:
build(docs-infra): configure region parser to allow `.mjs`

This commit updates the necessary config settings to allow `.mjs` files to be able to be parsed by the region parser. We configure the parser to use `inline-c-only` so that comments are generated in like the below

```
// #docregion
```

rather than

```
/* #docregion */
```

atscott commented:
This PR was merged into the repository by commit 36388b3ab23578151d7edad5106a26af69bcb45e.
